File: contests/FACENS-2024/solves/lousa.py
import math
import sys

input = sys.stdin.readline

while True:
	# A = largura
	# B = altura
	# C = D1
	# D = D2
	A, B, C, D = map(int, input().split())
	if (A, B, C, D) == (0, 0, 0, 0):
		break
		
	try:
		cosAngle1 = (math.pow(C, 2)) + (math.pow(A, 2)) - (math.pow(D, 2))
		cosAngle1 = cosAngle1 / (2 * C * A)
		
		angleValue = math.acos(cosAngle1)
		
		angleSin = math.sin(angleValue)
		angleCos = math.cos(angleValue)

		x = round(angleCos * C)
		y = round(angleSin * C)
				
		if (0 <= x <= A) and (0 <= y <= B):
			print(f"{x} {y}")
		else:
			print("DEFEITO")
	except:
		print("DEFEITO")

File: contests/FACENS-2024/solves/lousa.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void) {
	// A = largura
	// B = altura
	// C = distância até o sensor 1
	// D = distância até o sensor 2
	int a, b, c, d;
	
	while(1) {
		scanf("%d %d %d %d", &a, &b, &c, &d);
		if (a == 0 && b == 0 && c == 0 && d == 0) break;
		
		if (fabs(c - d) > a || (c + d) < a) {
			printf("DEFEITO\n");
			continue;
		}
		
		double angleCos = (a*a + c*c - d*d) / (2.0 * a * c);
		
		if (angleCos < -1) angleCos = -1;
		if (angleCos > 1) angleCos = 1;
		
		double angleValue = acos(angleCos);
		
		double x = c * cos(angleValue);
		double y = c * sin(angleValue);
		
		if (x < 0 || x > a || y < 0 || y > b) {
			printf("DEFEITO\n");
		}
		else {
			printf("%d %d\n", (int)round(x), (int)round(y));
		}
	}	
	
	return 0;
}
